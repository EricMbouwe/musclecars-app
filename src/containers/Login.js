/* eslint-disable jsx-a11y/label-has-associated-control */
import { useState } from 'react';
import PropTypes from 'prop-types';
import { signIn } from '../actions/actionCreator';

const Login = ({ dispatch }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(signIn(email, password));
  };

  return (
    <div>
      <h2>SIGN IN</h2>

      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <label htmlFor="name">Email</label>
          <input
            type="text"
            id="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </div>
        <div className="form-group">
          <label htmlFor="name">Password</label>
          <input
            type="text"
            id="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <button type="submit">login</button>
      </form>
    </div>
  );
};

Login.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

export default Login;
