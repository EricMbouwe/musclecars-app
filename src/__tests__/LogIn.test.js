import { render, screen } from '@testing-library/react';
import React from 'react';
import {
  BrowserRouter, Redirect, Route, Switch,
} from 'react-router-dom';
import '@testing-library/jest-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Login from '../containers/Login';

const initStore = { user: { } };
const mockStore = configureStore();
const store = mockStore(initStore);

let renderReadyComponent;
beforeEach(() => {
  renderReadyComponent = (
    <BrowserRouter>
      <Provider store={store}>
        <Redirect to="/second-page" />
        <Login />
        <Switch>
          <Route exact path="/" render={() => <>Main Page</>} />
          <Route exact path="/second-page" render={() => <div>Second Page</div>} />
        </Switch>
      </Provider>
    </BrowserRouter>
  );
});

describe('<Login />', () => {
  describe('If username exists', () => {
    it('redirect to \'/\' ', () => {
      initStore.user.username = 'mockName';
      render(renderReadyComponent);

      expect(screen.getByText(/Main Page/i)).toBeInTheDocument();
      expect(screen.queryByText(/Second Page/i)).not.toBeInTheDocument();
    });

    it('renders correctly', () => {
      initStore.user.username = 'mockName';
      const renderedContainer = render(renderReadyComponent);
      expect(renderedContainer).toMatchSnapshot();
    });
  });

  describe('If username does not exist', () => {
    it('renders Login Form if username is empty', () => {
      initStore.user.username = '';
      render(renderReadyComponent);

      expect(screen.getByText(/Second Page/i)).toBeInTheDocument();
      expect(screen.getByText(/Log In/i)).toBeInTheDocument();
      expect(screen.queryByText(/Main Page/i)).not.toBeInTheDocument();
    });

    it('renders correctly', () => {
      initStore.user.username = '';
      const renderedContainer = render(renderReadyComponent);

      expect(renderedContainer).toMatchSnapshot();
    });
  });
});
