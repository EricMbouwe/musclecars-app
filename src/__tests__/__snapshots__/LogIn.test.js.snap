// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> If username does not exist renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto"
      >
        <h2
          class="mt-5"
        >
          SIGN IN
        </h2>
        <form
          class="text-left p-8 border shadow-sm md:max-w-md mx-auto my-4 rounded-md"
        >
          <div
            class="form-group mt-3 flex flex-col"
          >
            <label
              class=""
              for="name"
            >
              Email
            </label>
            <input
              class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
              id="email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group my-3 flex flex-col"
          >
            <label
              for="name"
            >
              Password
            </label>
            <input
              class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
              id="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-green-100 text-base font-medium text-gray-700 hover:bg-green-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500 sm:mt-0 sm:w-auto sm:text-sm"
            type="submit"
          >
            login
          </button>
        </form>
      </div>
      <div>
        Second Page
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto"
    >
      <h2
        class="mt-5"
      >
        SIGN IN
      </h2>
      <form
        class="text-left p-8 border shadow-sm md:max-w-md mx-auto my-4 rounded-md"
      >
        <div
          class="form-group mt-3 flex flex-col"
        >
          <label
            class=""
            for="name"
          >
            Email
          </label>
          <input
            class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
            id="email"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group my-3 flex flex-col"
        >
          <label
            for="name"
          >
            Password
          </label>
          <input
            class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
            id="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-green-100 text-base font-medium text-gray-700 hover:bg-green-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500 sm:mt-0 sm:w-auto sm:text-sm"
          type="submit"
        >
          login
        </button>
      </form>
    </div>
    <div>
      Second Page
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Login /> If username exists renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto"
      >
        <h2
          class="mt-5"
        >
          SIGN IN
        </h2>
        <form
          class="text-left p-8 border shadow-sm md:max-w-md mx-auto my-4 rounded-md"
        >
          <div
            class="form-group mt-3 flex flex-col"
          >
            <label
              class=""
              for="name"
            >
              Email
            </label>
            <input
              class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
              id="email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group my-3 flex flex-col"
          >
            <label
              for="name"
            >
              Password
            </label>
            <input
              class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
              id="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-green-100 text-base font-medium text-gray-700 hover:bg-green-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500 sm:mt-0 sm:w-auto sm:text-sm"
            type="submit"
          >
            login
          </button>
        </form>
      </div>
      <div>
        Second Page
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto"
    >
      <h2
        class="mt-5"
      >
        SIGN IN
      </h2>
      <form
        class="text-left p-8 border shadow-sm md:max-w-md mx-auto my-4 rounded-md"
      >
        <div
          class="form-group mt-3 flex flex-col"
        >
          <label
            class=""
            for="name"
          >
            Email
          </label>
          <input
            class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
            id="email"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group my-3 flex flex-col"
        >
          <label
            for="name"
          >
            Password
          </label>
          <input
            class="p-1 border border-gray-300 text-gray-500 focus:ring-green-200"
            id="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-green-100 text-base font-medium text-gray-700 hover:bg-green-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-green-500 sm:mt-0 sm:w-auto sm:text-sm"
          type="submit"
        >
          login
        </button>
      </form>
    </div>
    <div>
      Second Page
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
